### PR TITLE
[KOGITO-187] Adding support for jvm builds

### PR DESCRIPTION
**Pull request by ricardozanini**
See:
https://issues.jboss.org/browse/KOGITO-187

In this PR:

Small change on the build to support JVM builds and runtimes. Waiting for the images to get ready to run smoke tests on the examples using this new approach.

TODO:
~1. Add an option to the CLI to deploy services using JVM builds~
~2. Run smoke tests using CLI and an OpenShift cluster to deploy the new service with JVM build~

Many thanks for submiting your Pull Request :heart:! 

Please make sure that your PR meets the following requirements:

- [x] You have read the [contributors guide](CONTRIBUTING.MD#sending-a-pull-request)
- [x] Pull Request title is properly formatted: `[KOGITO-XYZ] Subject`
- [x] Pull Request contains link to the JIRA issue
- [x] Pull Request contains description of the issue
- [x] Pull Request does not include fixes for issues other than the main ticket
- [x] Your feature/bug fix has a unit test that verifies it
- [x] You've tested the new feature/bug fix in an actual OpenShift cluster

### CODE REVIEW

**Comment by spolti** [comment]
0.4.0 stating images will be available soon.

File: pkg/controller/kogitoapp/builder/build_config.go [hunk 1]
```diff
@@ -4,21 +4,29 @@ import (
...
+	// BuildTypeRuntime will create a image with a Kogito Service available
+	BuildTypeRuntime BuildType = "runtime"
+	// BuildTypeRuntimeJvm will create a image with JRE installed to run the Kogito Service
+	BuildTypeRuntimeJvm BuildType = "runtime-jvm"
 	// LabelKeyBuildType is the label key to identify the build type
 	LabelKeyBuildType = "buildtype"
 )
 
+const (
+	kindImageStreamTag = "ImageStreamTag"
```
Comment (spolti) [inline pkg/controller/kogitoapp/builder/build_config.go]: why is this necessary? you can inherit this type from k8s or origin, or is it not possible?

Comment (ricardozanini) [reply pkg/controller/kogitoapp/builder/build_config.go]: We don't have the `Kind` in string format, just the type in the `openshift/api`. To use as string we should rely on reflection:

https://github.com/openshift/api/blob/master/image/v1/register.go

Do you know any place where those Kinds are available as `string`? This can come in handy for a nice refactor to our API.

Comment (spolti) [reply pkg/controller/kogitoapp/builder/build_config.go]: There's something similar on 
 

> "github.com/openshift/origin/pkg/oc/cli/create".create.ImageStreamTagRecommendedName

But not sure if make sense to use it.

File: README.md [hunk 2]
```diff
@@ -63,9 +63,31 @@ $ oc create -f deploy/crs/app_v1alpha1_kogitoapp_cr.yaml
...
+
+```bash
+$ kogito deploy-service example-quarkus https://github.com/kiegroup/kogito-examples/ --context-dir=drools-quarkus-example --native
+```
+
+#### Troubleshoting 
+
+If you don't see any builds running nor any resources created in the namespace, try to take a look at the Kogito Operator log.
+
+To look at the operator logs, first identify where it's deployed:
```
Comment (mramendi) [inline README.md]: "Where the operator is deployed"? Or what is "it"?

Comment (ricardozanini) [reply README.md]: I was using "it" to not have to repeat "the operator". But if you found that confusing, I'm fixing! :D

**Comment by ricardozanini** [comment]
@sutaakar @xiezhang7 could you guys please take a look?

Many thanks!
